File: Solutions/1664. Ways to Make a Fair Array/1664. Ways to Make a Fair Array.go
func waysToMakeFair(nums []int) int {
    l,r,res:=[2]int{},[2]int{},0
    for i:=0;i<len(nums);i+=1{
        r[i%2]+=nums[i]
    }
    for i:=0;i<len(nums);i+=1{
        r[i%2]-=nums[i]
        if x,y:=r[0]+l[1],r[1]+l[0];x==y{
            res+=1
        }
        l[i%2]+=nums[i]
    }
    return res
}

File: Solutions/1664. Ways to Make a Fair Array/1664. Ways to Make a Fair Array.java
class Solution {
    public int waysToMakeFair(int[] nums) {
        int[] l=new int[2],r=new int[2];
        int res=0;
        for(int i=0;i<nums.length;++i)
            r[i%2]+=nums[i];
        for(int i=0;i<nums.length;++i){
            r[i%2]-=nums[i];
            if(r[0]+l[1]==r[1]+l[0])
                ++res;
            l[i%2]+=nums[i];
        }
        return res;
    }
}